From a28abff2d2bb9419f75af25bc416056b65dc2b2a Mon Sep 17 00:00:00 2001
From: Hendrik Hagendorn <finnq@lineageos.org>
Date: Sun, 15 Jan 2017 08:31:13 +0100
Subject: [PATCH 13/15] LineageParts: Bring back and refactor battery icon
 options

* Use the icon blacklist for hiding
* Combine options for hiding status bar icons in system tuner settings

Change-Id: Id8c6df982192cfb54350c9cfe8d8ca4983dd7972
---
 res/values/arrays.xml                                 |  5 ++---
 res/xml/status_bar_settings.xml                       |  2 --
 .../lineageparts/statusbar/StatusBarSettings.java     | 11 ++---------
 3 files changed, 4 insertions(+), 14 deletions(-)

diff --git a/res/values/arrays.xml b/res/values/arrays.xml
index 79b0149..f958922 100644
--- a/res/values/arrays.xml
+++ b/res/values/arrays.xml
@@ -276,10 +276,9 @@
 
     <string-array name="status_bar_battery_style_values" translatable="false">
         <item>0</item>
-        <item>5</item>
+        <item>1</item>
         <item>2</item>
-        <item>6</item>
-        <item>4</item>
+        <item>3</item>
     </string-array>
 
     <string-array name="status_bar_battery_percentage_entries" translatable="false">
diff --git a/res/xml/status_bar_settings.xml b/res/xml/status_bar_settings.xml
index 4ef5fbf..a5525f9 100644
--- a/res/xml/status_bar_settings.xml
+++ b/res/xml/status_bar_settings.xml
@@ -83,7 +83,6 @@
             android:defaultValue="2" />
     </PreferenceCategory>
 
-<!--
     <PreferenceCategory
         android:title="@string/status_bar_battery_title">
 
@@ -106,7 +105,6 @@
             android:defaultValue="0" />
 
     </PreferenceCategory>
--->
 
     <PreferenceCategory
         android:title="@string/status_bar_brightness_category">
diff --git a/src/org/lineageos/lineageparts/statusbar/StatusBarSettings.java b/src/org/lineageos/lineageparts/statusbar/StatusBarSettings.java
index 7f7e843..0711fd8 100644
--- a/src/org/lineageos/lineageparts/statusbar/StatusBarSettings.java
+++ b/src/org/lineageos/lineageparts/statusbar/StatusBarSettings.java
@@ -43,8 +43,7 @@ public class StatusBarSettings extends SettingsPreferenceFragment
     private static final String STATUS_BAR_SHOW_BATTERY_PERCENT = "status_bar_show_battery_percent";
     private static final String STATUS_BAR_QUICK_QS_PULLDOWN = "qs_quick_pulldown";
 
-    private static final int STATUS_BAR_BATTERY_STYLE_HIDDEN = 4;
-    private static final int STATUS_BAR_BATTERY_STYLE_TEXT = 6;
+    private static final int STATUS_BAR_BATTERY_STYLE_TEXT = 3;
     private static final int PULLDOWN_DIR_NONE = 0;
     private static final int PULLDOWN_DIR_RIGHT = 1;
     private static final int PULLDOWN_DIR_LEFT = 2;
@@ -70,14 +69,12 @@ public class StatusBarSettings extends SettingsPreferenceFragment
         mStatusBarClockCategory =
                 (PreferenceCategory) getPreferenceScreen().findPreference(CATEGORY_CLOCK);
 
-/*
         mStatusBarBatteryShowPercent =
                 (LineageSystemSettingListPreference) findPreference(STATUS_BAR_SHOW_BATTERY_PERCENT);
         mStatusBarBattery =
                 (LineageSystemSettingListPreference) findPreference(STATUS_BAR_BATTERY_STYLE);
         mStatusBarBattery.setOnPreferenceChangeListener(this);
         enableStatusBarBatteryDependents(mStatusBarBattery.getIntValue(2));
-*/
 
         mQuickPulldown =
                 (LineageSystemSettingListPreference) findPreference(STATUS_BAR_QUICK_QS_PULLDOWN);
@@ -128,18 +125,14 @@ public class StatusBarSettings extends SettingsPreferenceFragment
         int value = Integer.parseInt((String) newValue);
         if (preference == mQuickPulldown) {
             updateQuickPulldownSummary(value);
-/*
         } else if (preference == mStatusBarBattery) {
             enableStatusBarBatteryDependents(value);
-*/
         }
         return true;
     }
 
     private void enableStatusBarBatteryDependents(int batteryIconStyle) {
-        mStatusBarBatteryShowPercent.setEnabled(
-                batteryIconStyle != STATUS_BAR_BATTERY_STYLE_HIDDEN
-                && batteryIconStyle != STATUS_BAR_BATTERY_STYLE_TEXT);
+        mStatusBarBatteryShowPercent.setEnabled(batteryIconStyle != STATUS_BAR_BATTERY_STYLE_TEXT);
     }
 
     private void updateQuickPulldownSummary(int value) {
-- 
2.17.1

